Add container-suseconnect (#39)
SUSE/doc-unversioned: container-guide/adoc/containers-bci.adoc

All the described commands use the `--non-interactive` flag to skip confirmations, since you cannot approve these manually during container builds. Keep in mind that you must use the flag with any command that modifies the system. Also note that `--non-interactive` is not a "yes to all" flag. Instead, `--non-interactive` confirms what is considered to be the intention of the user. For example, an installation command with the `--non-interactive` option fails if it needs to import new repository signing keys, as that is something that the user must verify themselves.

[[sec-bci-suseconnect]]
==== Using container-suseconnect with {bcia}s

https://github.com/SUSE/container-suseconnect[`container-suseconnect`]
is a plugin available in all {bcia}s that ship with
Zypper. When the plugin detects the host's {sles}
registration credentials, it uses them to give the container access the
SUSE Linux Enterprise repositories. This includes additional modules and
previous package versions that are not part of the free `SLE_BCI`
repository.

[[sec-bci-suseconnect-howto]]
=== Using container-suseconnect

> If you are running a registered {slsa} system with {docker},

Reviewer: `slsa`?

> `container-suseconnect` automatically detects and uses the subscription,

Reviewer: ```suggestion
`container-suseconnect` automatically discovers your credentials and uses the subscription,
```

Otherwise, copy both the `/etc/zypp/credentials.d/SCCcredentials` and
`/etc/SUSEConnect` files.

You can use the following command to obtain `SCCcredentials` (replace
`REGISTRATION_CODE` with your {scc} registration code).

{docker}:

[source,Shell]
----
docker run --rm registry.suse.com/suse/sle15:latest bash -c \
    "zypper -n in SUSEConnect; SUSEConnect --regcode REGISTRATION_CODE; \
     cat /etc/zypp/credentials.d/SCCcredentials"
----

{podman}:

[source,Shell]
----
podman run --rm registry.suse.com/suse/sle15:latest bash -c \
    "zypper -n in SUSEConnect; SUSEConnect --regcode REGISTRATION_CODE; \
     cat /etc/zypp/credentials.d/SCCcredentials"
----

{nerdctl}:

[source,Shell]
----
nerdctl run --rm registry.suse.com/suse/sle15:latest bash -c \
    "zypper -n in SUSEConnect; SUSEConnect --regcode REGISTRATION_CODE; \
     cat /etc/zypp/credentials.d/SCCcredentials"
----

If you are running a container based on a {bcia}, mount
`SCCcredentials` (and optionally `/etc/SUSEConnect`) in the correct
destination. The following example shows how to mount `SCCcredentials`
in the current working directory.

{docker}:

[source,Shell]
----

> docker run -v /path/to/SCCcredentials:/etc/zypp/credentials.d/SCCcredentials \
>     -it --pull=always registry.suse.com/bci/bci-base:latest

Reviewer: ```suggestion
docker run -v /path/to/SCCcredentials:/etc/zypp/credentials.d/SCCcredentials \
    --rm -it --pull=always registry.suse.com/bci/bci-base:latest
```

----

{podman}:

[source,Shell]
----

> podman run -v /path/to/SCCcredentials:/etc/zypp/credentials.d/SCCcredentials \
>     -it --pull=always registry.suse.com/bci/bci-base:latest

Reviewer: ```suggestion
podman run -v /path/to/SCCcredentials:/etc/zypp/credentials.d/SCCcredentials \
    --rm -it --pull=always registry.suse.com/bci/bci-base:latest
```

----

{nerdctl}:

[source,Shell]
----

> nerdctl run -v /path/to/SCCcredentials:/etc/zypp/credentials.d/SCCcredentials \
>     -it --pull=always registry.suse.com/bci/bci-base:latest

Reviewer: ```suggestion
nerdctl run -v /path/to/SCCcredentials:/etc/zypp/credentials.d/SCCcredentials \
    --rm -it --pull=always registry.suse.com/bci/bci-base:latest
```

----

Do not copy the `SCCcredentials` and `SUSEConnect` files into the
container image to avoid inadvertently adding them to the final image.
Use secrets instead, as they are only available to a single layer and
are not part of the built image. To do this, put a copy of
`SCCcredentials` (and optionally `SUSEConnect`) somewhere on the file

> system and modify the `RUN` instructions that invoke Zypper as follows:

Reviewer: Is zypper not also a replaceable?
```suggestion
system and modify the `RUN` instructions that invoke {zypper} as follows:
```

[source,Dockerfile]
----
FROM registry.suse.com/bci/bci-base:latest

RUN --mount=type=secret,id=SUSEConnect \
    --mount=type=secret,id=SCCcredentials \
    zypper -n in fluxbox
----

{docker} and {buildah} support mounting secrets via the `--secret` flag
as shown below.

{docker}:

[source,Shell]
----
docker build --secret=id=SCCcredentials,src=/path/to/SCCcredentials \
    --secret=id=SUSEConnect,src=/path/to/SUSEConnect .
----

{podman}:

[source,Shell]
----
buildah bud --layers --secret=id=SCCcredentials,src=/path/to/SCCcredentials \
    --secret=id=SUSEConnect,src=/path/to/SUSEConnect .
----

[[sec-bci-suseconnect-add-modules]]
==== Adding modules to a container or container image

`container-suseconnect` allows you to automatically add {slea} Modules to
a container or container image. What modules are added is determined by
the environment variable `ADDITIONAL_MODULES` that includes a
comma-separated list of the module names. In a `Dockerfile`, this

> is done using the `ENV` directive as follows:

Reviewer: ```suggestion
is achieved using the `ENV` directive as follows:
```

....
> docker build .

> Sending build context to {docker} daemon  2.048kB

Reviewer: This is a console output, it should not be influenced by our replacements.
```suggestion
Sending build context to Docker daemon  2.048kB
```
